18 Normalizr/src/model/api/student.ts

```diff
@@ -3,12 +3,14 @@ export class Student {
   gotActiveTraining: boolean;
   fullname: string;
   email: string;
+  countryId: number;
 
   public constructor() {
     this.id = -1;
     this.gotActiveTraining = false;
     this.fullname = "";
     this.email = "";
+    this.countryId = -1;
```

Comment:
Yep, if it's a component problem, let the component to deal with it.
On the other hand, I would use an interface too. I avoid factory methods unless you need to share some initialization logic that I don't see on this case.